src/database/migrations/0006-create-proposals.sql

```diff
@@ -0,0 +1,21 @@
+CREATE TABLE proposals (
+  id INTEGER PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+  external_id VARCHAR NOT NULL,
```

Comment:
It might be better to have the `varchar` fields follow the `integer` fields for packing efficiency. Or if the logical order as found in the unique index below is applicant_id then opportunity_id then external_id to make it consistent with that.